app/src/renderer/components/wallet/SendModal.vue

```diff
@@ -1,143 +1,88 @@
+    ref="actionModal"
+    :submit-fn="submitForm"
+    :validate="validateForm"
+    title="Send"
+    submission-error-prefix="Sending tokens failed"
+  >
     <tm-form-group
-      :error="$v.fields.denom.$error && $v.fields.denom.$invalid"
+      :error="$v.denom.$dirty && $v.denom.$invalid"
+      class="action-modal-form-group"
       field-id="send-denomination"
       field-label="Denomination"
     >
       <tm-field
         id="send-denomination"
-        v-model.number="$v.fields.denom.$model"
+        v-model.number="$v.denom.$model"
         type="text"
         readonly
       />
       <tm-form-msg
-        v-if="
-          $v.fields.denom.$error &&
-            $v.fields.denom.$invalid &&
-            !$v.fields.denom.required
-        "
+        v-if="$v.denom.$error && !$v.denom.required"
         name="Denomination"
         type="required"
       />
     </tm-form-group>
 
     <tm-form-group
-      :error="$v.fields.address.$error && $v.fields.address.$invalid"
+      :error="$v.address.$error && $v.address.$invalid"
+      class="action-modal-form-group"
       field-id="send-address"
       field-label="Send To"
     >
       <tm-field
         v-focus
         id="send-address"
-        v-model.number="$v.fields.address.$model"
+        v-model.number="$v.address.$model"
         type="text"
         placeholder="Address"
       />
       <tm-form-msg
-        v-if="$v.fields.address.$error && !$v.fields.address.required"
+        v-if="$v.address.$error && !$v.address.required"
         name="Address"
         type="required"
       />
       <tm-form-msg
-        v-else-if="
-          $v.fields.address.$error && !$v.fields.address.bech32Validate
-        "
+        v-else-if="$v.address.$error && !$v.address.bech32Validate"
         :body="bech32error"
         name="Address"
         type="bech32"
       />
     </tm-form-group>
-
     <tm-form-group
-      :error="$v.fields.address.$error && $v.fields.address.$invalid"
-      field-id="send-amount"
+      :error="$v.amount.$error && $v.amount.$invalid"
+      class="action-modal-form-group"
+      field-id="amount"
       field-label="Amount"
     >
       <tm-field
-        id="send-amount"
-        v-model.number="$v.fields.amount.$model"
-        type="number"
+        id="amount"
+        v-model="amount"
+        class="tm-field"
         placeholder="Amount"
+        type="number"
       />
       <tm-form-msg
-        v-if="$v.fields.amount.$error && !$v.fields.amount.required"
-        name="Amount"
-        type="required"
-      />
-      <tm-form-msg
-        v-if="
-          $v.fields.amount.$error &&
-            !$v.fields.amount.between &&
-            amount > 0 &&
-            balance > 0
-        "
-        :max="$v.fields.amount.$params.between.max"
-        :min="$v.fields.amount.$params.between.min"
-        name="Amount"
-        type="between"
-      />
-      <tm-form-msg
-        v-else-if="max === 0"
+        v-if="balance === 0"
```

Comment:
we should have this message on the action modal instead

Reply:
hmm the forms are highly different, I don't know if we could refactor this into the action modal

Reply:
then we'd have to add it to every modal that involves amount (can be done on another PR)

Reply:
https://github.com/cosmos/voyager/issues/1885